### PR TITLE
Message details: Hide "Add to contacts" button when contacts permission is missing

### PR DESCRIPTION
**Pull request by cketti**

### CODE REVIEW

File: core/android/common/src/main/kotlin/app/k9mail/core/android/common/contact/ContactRepository.kt [hunk 1]
```diff
@@ -10,6 +10,8 @@ interface ContactRepository {
     fun hasContactFor(emailAddress: EmailAddress): Boolean
 
     fun hasAnyContactFor(emailAddresses: List<EmailAddress>): Boolean
+
+    fun hasContactPermission(): Boolean
```
Comment (wmontwe) [inline core/android/common/src/main/kotlin/app/k9mail/core/android/common/contact/ContactRepository.kt:14]: You could use the `ContactPermissionResolver` directly, no need to also include it in the repository. This just broadens it's responsibility.

**Review by wmontwe — COMMENTED** [review]
I like that the contact permission check is extracted now.

File: core/android/common/src/main/kotlin/app/k9mail/core/android/common/contact/ContactRepository.kt (hunk 1 above)
Comment (cketti) [reply core/android/common/src/main/kotlin/app/k9mail/core/android/common/contact/ContactRepository.kt:14]: It felt like functionality a user of `ContactRepository` would sometimes need. But I'm not set on this approach. If you think `MessageDetailsViewModel` should use `ContactPermissionResolver` directly, I'll change it.

Comment (wmontwe) [reply core/android/common/src/main/kotlin/app/k9mail/core/android/common/contact/ContactRepository.kt:14]: It is better to use it directly

**Review by wmontwe — APPROVED** [review]
LGTM

**Pull request merged**